src/thorin/be/codegen.cpp

```diff
@@ -87,11 +88,12 @@ DeviceBackends::DeviceBackends(World& world, int opt, bool debug, std::string& f
         Continuation* imported = nullptr;
 
         static const auto backend_intrinsics = std::array {
-            std::pair { CUDA,   Intrinsic::CUDA   },
-            std::pair { NVVM,   Intrinsic::NVVM   },
-            std::pair { OpenCL, Intrinsic::OpenCL },
-            std::pair { AMDGPU, Intrinsic::AMDGPU },
-            std::pair { HLS,    Intrinsic::HLS    }
+            std::pair { CUDA,       Intrinsic::CUDA      },
+            std::pair { NVVM,       Intrinsic::NVVM      },
+            std::pair { OpenCL,     Intrinsic::OpenCL    },
+            std::pair { AMDGPU_HSA, Intrinsic::AMDGPUHSA },
+            std::pair { AMDGPU_PAL, Intrinsic::AMDGPUPAL },
```

Comment:
While not on the table right now, I have discussed the possibility of making backends configurable from the Thorin side directly. As in: pass arguments from the intrinsic to the back-end code, to configure things like subtargets and versions.

This seems like it will be the perfect candidate for that work once I get to it, since the most significant change here appears to be the target triple